### PR TITLE
feat(Slug): implement phase 3 updates, adjust token values

### LINKED ISSUE #15938: [AI] Phase 3: Update AI Slug with new visuals
We are simplifying the slug component. There will no longer be two kinds (default and hollow). Now there will be just one style icon and one style callout popover. Here is a list of the original slugs and which we are deprecating. 

Slug deprecation list v3

## New slug spec
[Figma file](https://www.figma.com/file/7eMfOAyRclD4LuDFwRAa3p/Carbon-for-AI-Sketches?type=design&node-id=2280%3A32647&mode=design&t=hH6LD5yVVCAhYdRZ-1) 
Slug - Standard - Specs v3

## New inline slug spe
[Figma file](https://www.figma.com/file/7eMfOAyRclD4LuDFwRAa3p/Carbon-for-AI-Sketches?type=design&node-id=2280%3A32647&mode=design&t=hH6LD5yVVCAhYdRZ-1) 
Slug - Inline - Specs v3

## New slug callout spec
[Figma file](https://www.figma.com/file/7eMfOAyRclD4LuDFwRAa3p/Carbon-for-AI-Sketches?type=design&node-id=2128%3A158573&mode=design&t=hH6LD5yVVCAhYdRZ-1)

Slug - Callout - Specs v3 - light
Slug - Callout - Specs v3

### LINKED ISSUE #15941: [AI] Phase 3: update AI token values
Phase 3 visuals require some token value changes. Below are listed only the tokens that need a value update. See [boxnote](https://ibm.box.com/s/x88jcrtnlbf4br2w08fup79xaqqywdn3) for full audit.

[Figma file](https://www.figma.com/file/JLbAqBUlc1IGt8pUe2C8MD/SANDBOX---Layering-model?type=design&node-id=1965%3A67676&mode=design&t=HNPXSP2YSdsjx2B6-1)

## Shared AI tokens

Token | White | Gray 10 | Gray 90 | Gray 100
------|--------|-------|-----------|---------
ai-inner-shadow | blue 50, 10% | blue 50, 10% | blue 50, 16% | blue 50, 16% 
ai-aura-start | blue 50, 5% | blue 50, 5% | blue 50, 10% | blue 50, 10%
ai-border-start | blue 30, 64% | blue 30, 64% | blue 30, 36% | blue 30, 36% 
ai-border-end | blue 40 | blue 40 | blue 50 | blue 50 
ai-drop-shadow | blue 60, 10% | blue 60, 10% | black, 28% | black, 28%
ai-aura-hover-start | blue 50, 32% | blue 50, 32% | blue 50, 40% | blue 50, 40%
ai-overlay | blue 100, 50% | blue 100, 50% | black, 50% | black, 50%

## AI popover tokens

Token | White | Gray 10 | Gray 90 | Gray 100
------|--------|-------|-----------|---------
ai-popover-background | white | white | gray 100 | gray 100 
ai-popover-shadow-outer-01 | blue 70, 6% | blue 70, 6% | black 12% | black 12%
ai-popover-shadow-outer-02 | black, 4% | black, 4% | black 8% | black 8%

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/15938
Closes https://github.com/carbon-design-system/carbon/issues/15941

This PR makes changes to the `Slug`, namely removing the `hollow` prop and setting this variant as the default. The gradient version of the `Slug` has been removed, as well as all associated styles/tokens. This also updates the `Slug` and the AI popover to use new token values that change the AI aura/gradient / other subtle effects. 

Screenshot 2024-03-14 at 10 41 42 AM

Screenshot 2024-03-14 at 10 42 15 AM

#### Changelog

**New**

- `$ai-popover-background` token, used in the `ai-popover-gradient` mixin instead of `$background`

**Changed**

 - The `kind` prop on `Slug` now only accepts `default` and `inline`. `default` now uses the `hollow` styles
 - All `Slug` variants should use the AI Popover
 - Token values have been updated to match the phase 3 spec
 - A few changes to the AI popover drop shadow / inner shadow values

**Removed**

- Removed hollow slug stories, hollow dot-type stories 
- Removed the following tokens:
  - `$slug-background`
  - `$slug-background-hover`
  - `$slug-gradient`
  - `$slug-gradient-hover`
  - `$slug-hollow-hover`
 - Removed the `dotType` prop on `Slug`
 - Removed `cds--slug--enabled` since all slug callouts now use the AI popover

#### Testing / Reviewing

Ensure there are no occurrences of `hollow` in the repo

### CODE REVIEW

File: packages/themes/src/g10.js [hunk 1]
```diff
@@ -205,28 +205,22 @@ export const toggleOff = gray50;
...
+export const aiBorderEnd = blue40;
+export const aiDropShadow = rgba(blue60, 0.1);
 export const aiAuraHoverBackground = blue10;
-export const aiAuraHoverStart = rgba(blue50, 0.4);
+export const aiAuraHoverStart = rgba(blue50, 0.32);
 export const aiAuraHoverEnd = rgba(white, 0);
 
 // AI Popover
+export const aiPopoverBackground = white;
 export const aiPopoverShadowOuter01 = rgba(blue70, 0.25);
```
Comment (aagonzales) [inline packages/themes/src/g10.js]:
```suggestion
export const aiPopoverShadowOuter01 = rgba(blue70, 0.06);
```

File: packages/themes/src/g10.js [hunk 2]
```diff
@@ -205,28 +205,22 @@ export const toggleOff = gray50;
...
 export const aiPopoverShadowOuter02 = rgba(black, 0.1);
```
Comment (aagonzales) [inline packages/themes/src/g10.js]:
```suggestion
export const aiPopoverShadowOuter02 = rgba(black, 0.04);
```

**Review by aagonzales — COMMENTED** [review]
Gray 10 looks good now and I doubled checked the other values/themes. Everything is good. We just need to update the caret tokens now and I think this will be ready for review. 

image

Token | White | Gray 10 | Gray 90 | Gray 100
------|--------|-------|-----------|---------
aiPopoverCaretCenter | `#A0C3FF` | `#A0C3FF` | `#4870B5` | `#4870B5`
aiPopoverCaretBottom | `blue40` | `blue40` | `blue50` | `blue50`
aiPopoverCaretBottomBackgroundActions | `#E9EFFA`  | `#E9EFFA` | `#1E283A` | `#1E283A`
aiPopoverCaretBottomBackground | `#EAF1FF` | `#EAF1FF` | `#202D45` | `#202D45`

**Review by alisonjoseph — APPROVED** [review]

**Comment by aagonzales** [comment]
@tw15egan Ok this is a long comment. During the Figma kit work, we noticed the gradient in the inputs was barely noticeable. 

image

But here, in the code examples, it was showing up better. Upon further investigation, I noticed the gradient was using `ai-inner-shadow` instead of `ai-aura-start` which is why it looked better.

image

We want to leave the value of `ai-aura-start` alone because it does look good in larger containers like tiles. 
image


**Conclusion** 
We need a _new token_ to use at the start of the gradients in inputs. My thinking is that we can consolidate with `$ai-aura-start-table` and essentially replace it with a broader name. `ai-aura-start-sm` since its purpose is to be used in smaller container sizes. 

Token | White | G10 | G90 | G100
------|-------|------|-----|------
`ai-aura-start-sm` | blue50, 0.16 | blue50, 0.16 | blue50, 0.16 | blue50, 0.16

image
image

**Review by aagonzales — APPROVED** [review]
Ok I think everything in here is good now! 🎉

**Review by aagonzales — COMMENTED** [review]
Actually, there is one more token change update from Vora if we can get it in here. Its about skeleton states again. Just the dark themes 

- skeleton-background: blue 40, 50% 
- skeleton-element: blue 40, 30%

**Comment by tw15egan** [comment]
@aagonzales I had to make a different bug fix with the AI Skeleton, so I bundled those tokens changes over in https://github.com/carbon-design-system/carbon/pull/15991

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**